backend/handlers/handlers.go

```diff
@@ -989,6 +992,14 @@ func (handlers *Handlers) getAccountSummary(_ *http.Request) (interface{}, error
 		return []chartEntry{}
 	}
 
+	// Even if we are still gathering data (exchange rates, block headers), we know we result
+	// already if there are no transactions. This avoids showing the user a message that we are
+	// gathering data, only to show nothing in the end.
+	if chartDataMissing && totalNumberOfTransactions == 0 {
```

Comment:
Or simply `if totalNumberOfTransactions == 0`?

Reply:
Then the log would not be as useful.

Reply:
A test would've been even more useful than a log message. Then you wouldn't need to log :)

Reply:
Mutually exclusive ;)

Unit tests I hope I'll get around to. https://github.com/digitalbitbox/bitbox-wallet-app/pull/1119 will help